SAI Release Notes for 1.6.5

kannankvs opened the pull request.

rlhui commented:
@DeanKalla, @michaelli10 - would you please help to review/approve this SAI 1.6 release notes, (saimacsec and saisystemport) are two main features added in this release. Thanks.

rlhui commented on doc/SAI_1.6.5_ReleaseNotes.md:
```diff
@@ -0,0 +1,539 @@
+17. saitypes
+
+
+### sah.h		
```
typo here

rlhui reviewed (commented):
@phshaikh , @bandaru-viswanath , @marian-pritsak , @AshokDaparthi , @JaiOCP , would you please help review/approve this release notes?

bandaru-viswanath commented on doc/SAI_1.6.5_ReleaseNotes.md:
````diff
@@ -0,0 +1,539 @@
+```
+
+###saitam.h
+Definition for metadata fragment has been enabled.  When there is insufficient space in the packet to add INT metadata for this hop (e.g. MTU would be exceeded), the device may remove the metadata from the packet, send a report to the collector, and insert its metadata before forwarding the packet. SAI_TAM_REPORT_ATTR_ENTERPRISE_NUMBER has been modified for the saitam file.
````
SAI_TAM_REPORT_ATTR_ENTERPRISE_NUMBER has been added as an attribute for the TAM Report object in the saitam.h file.

bandaru-viswanath commented on doc/SAI_1.6.5_ReleaseNotes.md:
```diff
@@ -0,0 +1,539 @@
+
+
+### saitunnel.h	
+Create and Set for Tunnel Attributes are added.
```
Support for P2P tunnel types added, in addition to existing P2MP mode. Additionally, some attributes are now allowed to be set after creation of the tunnel objects.

bandaru-viswanath commented on doc/SAI_1.6.5_ReleaseNotes.md:
```diff
@@ -0,0 +1,539 @@
+     */
+    SAI_TUNNEL_PEER_MODE_P2MP,
+
+} sai_tunnel_peer_mode_t;
```
I think the sai_tunnel.h has more diffs, than what is presented here.

kannankvs replied:
Done

michaelli10 commented on doc/SAI_1.6.5_ReleaseNotes.md:
```diff
@@ -0,0 +1,539 @@
+SAI_PORT_ATTR_INGRESS_MACSEC_ACL
+SAI_PORT_ATTR_EGRESS_MACSEC_ACL
+SAI_PORT_ATTR_MACSEC_PORT_LIST
+
```
Will you add the system port changes (and other saiport.h changes) from PR1080?https://github.com/opencomputeproject/SAI/pull/1081

    /**
     * @brief System port for the port
     *
     * @type sai_object_id_t
     * @flags READ_ONLY
     * @objects SAI_OBJECT_TYPE_SYSTEM_PORT
     */
    SAI_PORT_ATTR_SYSTEM_PORT,

michaelli10 commented on doc/SAI_1.6.5_ReleaseNotes.md:
```diff
@@ -0,0 +1,539 @@
+7)Platform specific device register write access. This API provides platform adaption functionality to access device registers from driver. This is mandatory to pass as attribute to sai_create_switch when driver implementation does not support register access by device file system directly.<br>
+8)Switch MDIO read API - Provides read access API for devices connected to MDIO from NPU SAI.<br>
+9)Switch MDIO write API - Provides write access API for devices connected to MDIO from NPU SAI.<br>
+
```
Can you add an item about the new system port related attributes?

SAI_SWITCH_ATTR_SWITCH_ID
SAI_SWITCH_ATTR_MAX_SYSTEM_CORES
SAI_SWITCH_ATTR_SYSTEM_PORT_CONFIG_LIST
SAI_SWITCH_ATTR_NUMBER_OF_SYSTEM_PORTS
SAI_SWITCH_ATTR_SYSTEM_PORT_LIST
SAI_SWITCH_ATTR_NUMBER_OF_FABRIC_PORTS
SAI_SWITCH_ATTR_FABRIC_PORT_LIST

michaelli10 reviewed (commented):
Does it make sense to separate the changes by feature similar to SAI 1.5 release notes for context?  Also, adding the PR references would be helpful.  https://github.com/opencomputeproject/SAI/blob/v1.6/doc/SAI_1.5_ReleaseNotes.md

DeanKalla commented:
> @DeanKalla, @michaelli10 - would you please help to review/approve this SAI 1.6 release notes, (saimacsec and saisystemport) are two main features added in this release. Thanks.

saimacsec changes look good to me

rlhui commented:
> > @DeanKalla, @michaelli10 - would you please help to review/approve this SAI 1.6 release notes, (saimacsec and saisystemport) are two main features added in this release. Thanks.
> 
> saimacsec changes look good to me

@DeanKalla - thanks. would you please approve if it's good to you?

kannankvs commented on doc/SAI_1.6.5_ReleaseNotes.md:
```diff
@@ -0,0 +1,539 @@
+
+### saisystemport.h
+
+System port creation can be accomplished by calling switch_create() with the list of system ports to be created using SAI_SWITCH_ATTR_SYSTEM_PORT_CONFIG_LIST or creating them individually using the create_system_port() function in the system port APIs. Although it is expected that all local ports be created in switch_create().
```
Done

kannankvs commented on doc/SAI_1.6.5_ReleaseNotes.md:
````diff
@@ -0,0 +1,539 @@
+
+```
+
+ /** Default packet action is forward */
````
Done

kannankvs commented on doc/SAI_1.6.5_ReleaseNotes.md:
```diff
@@ -0,0 +1,539 @@
+     * @isvlan false
+     * @default 0x8100
+     */
+    SAI_LAG_ATTR_TPID,
```
Yes, as its included in the new version

rlhui commented:
> Does it make sense to separate the changes by feature similar to SAI 1.5 release notes for context? Also, adding the PR references would be helpful. https://github.com/opencomputeproject/SAI/blob/v1.6/doc/SAI_1.5_ReleaseNotes.md

@kannankvs  - yes, please add that the below features can be supported with this SAI release:

-	Gearbox/Phy/MACsec Support
-	VOQ System
-	MPLS Segment Routing Enhancements
-	Fine-grained and ordered ECMP types
-	ACL GRE key match
-	ACL support to match TAM/INT packets
-	Bulk api for fdb entry
-	NRZ and PAM-4 in meta dictionary
-	Enable single chip data plane testing without external TG
-	TPID configuration
-	IPFix enhancement
-	Add support for matching any INT packets
-	Metadata enhancements

rlhui commented:
@michaelli10  - please check if your comments are addressed? Thanks.

rlhui commented:
@phshaikh , @marian-pritsak , @AshokDaparthi , @JaiOCP  - would you please help to review/sign off this RN? Thanks.

kannankvs commented:
SONiC 202006 is using SAI 1.6.3 and hence we shall use only the 1.6.3 release notes. We have addressed all the comments given in this PR in the 1.6.3 release notes. Please refer [PR1131](https://github.com/opencomputeproject/SAI/pull/1131) and sign off in that PR1131. We are closing this PR1134 with no further action.
